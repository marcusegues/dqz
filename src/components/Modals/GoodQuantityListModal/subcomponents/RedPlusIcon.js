// @flow
import React from 'react';
// $FlowFixMe
import { TouchableOpacity } from 'react-native';
import Entypo from '@expo/vector-icons/Entypo';
import { MAIN_RED } from '../../../../styles/colors';
import { moderateScale } from '../../../../styles/Scaling';

type RedPlusIconProps = {
  onPress: () => void,
};

export const RedPlusIcon = ({ onPress }: RedPlusIconProps) => (
  <TouchableOpacity onPress={onPress}>
    <Entypo name="circle-with-plus" size={moderateScale(36)} color={MAIN_RED} />
  </TouchableOpacity>
);
