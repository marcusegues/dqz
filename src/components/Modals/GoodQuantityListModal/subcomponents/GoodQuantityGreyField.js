// @flow
import React from 'react';
import { translate } from 'react-i18next';
import type { ComponentType } from 'react';
// $FlowFixMe
import { Text, View } from 'react-native';
import { CardHeaderSubText } from '../../../QuestionAnswer/Cards/subcomponents/CardHeaderSubText';
import { moderateScale, verticalScale } from '../../../../styles/Scaling';
import { MAIN_RED } from '../../../../styles/colors';
import { RedPlusIcon } from './RedPlusIcon';

const ownStyles = {
  container: {
    flex: 1,
    width: '90%',
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#F5F5F5',
    marginBottom: 0,
  },
  topText: {
    alignSelf: 'flex-start',
    paddingTop: verticalScale(16),
    paddingBottom: verticalScale(35),
  },
  plusIconText: {
    fontSize: moderateScale(14),
    color: MAIN_RED,
    fontFamily: 'roboto_medium',
    marginTop: verticalScale(10),
    marginBottom: verticalScale(30),
  },
};

type GoodQuantityGreyFieldProps = {
  topText: string,
  plusIconText: string,
  onPress: () => void,
};

const GoodQuantityGreyFieldInner = ({
  topText,
  plusIconText,
  onPress,
}: GoodQuantityGreyFieldProps) => (
  <View style={ownStyles.container}>
    <CardHeaderSubText text={topText} style={ownStyles.topText} />
    <View style={{ alignItems: 'center' }}>
      <RedPlusIcon onPress={() => onPress()} />
      <Text style={ownStyles.plusIconText}>{plusIconText.toUpperCase()}</Text>
    </View>
  </View>
);

export const GoodQuantityGreyField = (translate([
  'mainCategories',
  'categories',
])(GoodQuantityGreyFieldInner): ComponentType<GoodQuantityGreyFieldProps>);
