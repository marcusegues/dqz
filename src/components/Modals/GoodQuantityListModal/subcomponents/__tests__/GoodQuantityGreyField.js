// @flow
import React from 'react';
// $FlowFixMe
import { TouchableOpacity, TouchableWithoutFeedback } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { GoodQuantityGreyField } from '../GoodQuantityGreyField';
import { RedPlusIcon } from '../RedPlusIcon';

const wrapper = shallow(
  <GoodQuantityGreyField
    topText="topText"
    plusIconText="plusIconText"
    onPress={() => {}}
  />
);
// console.log(wrapper.debug());

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const clickFn = jest.fn();

describe('InfoIcon Icon TestSuite', () => {
  test('InfoIcon component renders correctly', () => {
    const tree = renderer
      .create(
        <GoodQuantityGreyField
          topText="topText"
          plusIconText="plusIconText"
          onPress={clickFn}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('we can click it', () => {
    const component = shallow(
      <GoodQuantityGreyField
        topText="topText"
        plusIconText="plusIconText"
        onPress={() => {}}
      />
    );
    // console.log(component.debug());

    // component
    //   .find(RedPlusIcon)
    //   .find(TouchableOpacity)
    //   .forEach(child => {
    //     // component.find(TouchableOpacity).forEach(child => {
    //     child.simulate('press');
    //   });
    // expect(clickFn).toHaveBeenCalled();
    // component.find(TouchableOpacity).simulate('press');
    // expect(clickFn).toBeCalled();
    // expect(clickFn).toBeCalledWith({
    //   type: "NAVIGATE",
    //   screen: "AppInfo"
    // });
  });
  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    const wrapper = shallow(
      <GoodQuantityGreyField
        topText="topText"
        plusIconText="plusIconText"
        onPress={onButtonClick}
      />
    );
    // const wrapper = shallow(<Foo onButtonClick={onButtonClick} />);
    wrapper.find(TouchableOpacity).simulate('click');
    // expect(onButtonClick).toBeCalled();
    // expect(onButtonClick.calledOnce).to.equal(true);
  });
});
