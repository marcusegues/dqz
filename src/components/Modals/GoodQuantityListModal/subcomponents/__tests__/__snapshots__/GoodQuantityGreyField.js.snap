// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIcon Icon TestSuite InfoIcon component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "marginBottom": 0,
      "width": "90%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "#9B9B9B",
        "fontFamily": "roboto_regular",
        "fontSize": 14,
        "paddingBottom": 68.66176470588235,
        "paddingHorizontal": 31.388235294117646,
        "paddingTop": 31.388235294117646,
      }
    }
  >
    topText
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#DC0018",
          "fontFamily": "roboto_medium",
          "fontSize": 22,
          "marginBottom": 58.85294117647059,
          "marginTop": 19.61764705882353,
        }
      }
    >
      PLUSICONTEXT
    </Text>
  </View>
</View>
`;
